src/syft/ast/klass.py

```diff
@@ -232,6 +232,11 @@ def __iter__(self: Any) -> Iterable:
 
     attr_name = "__iter__"
     iter_target = attrs[attr_name]
+
+    # skip if __iter__ has already been wrapped
+    if "__qual_name__" in dir(iter_target) and "wrap_iter" in iter_target.__qualname__:
```

Comment:
`
qual_name = getattr(iter_target, "__qual_name__", None)
if qual_name and "wrap_iter" in qual_name:
    return
`